sbin/submitty_autograding_shipper.py

```diff
@@ -400,6 +536,20 @@ def get_job(my_name,which_machine,my_capabilities,which_untrusted,overall_lock):
     overall_lock.acquire()
     folder= INTERACTIVE_QUEUE
 
+    # Our first priority is to perform any awaiting VCS checkouts
+    # Grab all the VCS files currently in the folder...
+    vcs_files = [str(f) for f in Path(folder).glob('VCS__*')]
+    for f in vcs_files:
+        vcs_file = f[len(folder)+1:]
+        no_vcs_file = f[len(folder)+1+5:]
+        # do the checkout
+        updated_obj = checkout_vcs_repo(folder+"/"+vcs_file)
+        # save the regular grading queue file
+        with open(os.path.join(folder,no_vcs_file), "w") as queue_file:
+            json.dump(updated_obj, queue_file)
+        # cleanup the vcs queue file
+        os.remove(folder+"/"+vcs_file)
```

Comment:
I think this needs to be rethought?

My reading of this is entire function with this change is:
1. acquire lock (causing all other workers to wait till this worker gets a job)
2. get the list of all waiting VCS checkouts to do
3. Do the all of those VCS checkouts
4. Get the job to operate on in the queue directory
5. Release the lock and return

Where step 3 can last for quite a while if there's a long list of repositories waiting or if the repositories are large or the network (or some combination of the three) meaning that we could have all other workers just wait. Except during this potentially long wait, we get even more repositories, so the next worker is also slow to work, and so on and so forth and our throughput is wrecked.

Reply:
Yeah, I didn't want to agonize too much since we are supposedly going to re-do the scheduler completely.  I settled on this imperfect solution for now because practically the load never gets this high.